packages/shared/src/components/drawers/NavDrawer.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from 'react';
+import {
+  Drawer,
+  DrawerPosition,
+  DrawerRef,
+  DrawerWrapperProps,
+} from './Drawer';
+import { NavDrawerItem, NavItemProps } from './NavDrawerItem';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
+import { ArrowIcon } from '../icons';
+import classed from '../../lib/classed';
+
+interface NavDrawerProps {
+  drawerProps: Omit<DrawerWrapperProps, 'children'>;
+  items: NavItemProps[];
+  header?: string;
+}
+
+const NavDrawerHeader = classed(
+  'div',
+  'flex items-center gap-3 px-4 border-b border-theme-divider-tertiary h-12',
+);
+
+const NavDrawerContent = classed('div', 'flex flex-col px-4 py-5');
+
+const NavHeading = classed('h2', 'typo-body font-bold');
+
+export function NavDrawer({
+  drawerProps,
+  header,
+  items,
+}: NavDrawerProps): ReactElement {
+  const ref = React.useRef<DrawerRef>();
+  const {
+    position,
+    displayCloseButton,
+    closeOnOutsideClick,
+    ...otherDrawerProps
+  } = drawerProps;
+
+  return (
+    <Drawer
+      {...otherDrawerProps}
+      closeOnOutsideClick={false}
+      displayCloseButton={displayCloseButton}
+      position={position ?? DrawerPosition.Left}
+      ref={ref}
+      role="menu"
+      className={{
+        drawer: 'bottom-0 top-0 !max-h-none !rounded-none',
```

Comment:
For drawers coming from the sides, these classes should be coming from the Drawer component itself.

We can modify the `drawerPositionToClassName` to include these classes, along with the removal of max height and only applied for top or bottom drawer.

Reply:
This is more of a `full-screen` drawer edge case. Maybe I should add it as a prop to the drawer itself, which will add all of the positioning classes for us.

We still might want a drawer to animate from the side, but not cover the whole screen...

Reply:
Yeah, that is true. We may want full screen but we can still decide the origin of the animation.

Reply:
Added an `isFullScreen` prop to the `Drawer` and refactored the styling accordingly. Now we don't need to supply these anymore.